utils/sl/check_script.py

```diff
@@ -268,59 +250,45 @@ def check_script(fname, *args, **kw):
-    error_count = parse_report(log_file)
-    print("")
-    return error_count
+    parse_report(log_file)
 #pylint: enable=too-many-branches
 
 def main():
     """Run the actual code in a function"""
     parser = argparse.ArgumentParser("Check a Python script for errors")
-    parser.add_argument("fname", metavar='FILENAME', type=str, nargs='?',
+    parser.add_argument("fname", metavar='FILENAME', type=str, nargs='*',
                         default=None, help="Filename of script to check")
     parser.add_argument("-w", dest='wrap', action='store_true',
                         help='Wrap the SCons script before checking')
+    parser.add_argument('-x', dest='exclude', help='Path to exclude')
     parser.add_argument("-s", dest='self_check', action='store_true',
                         help='Perform a self check')
-    parser.add_argument("-p3", dest='P3', action='store_true',
-                        help='Perform check using python3 (default)')
 
     args = parser.parse_args()
 
-    error_count = 0
-
-    pylint_rc = create_rc("pylint.rc")
     pylint3_rc = create_rc("pylint3.rc")
 
     if args.self_check:
-        print("Checking SCons")
-        error_count += check_script("SCons",
-                                    "-d", "too-few-public-methods",
-                                    "-d", "too-many-public-methods",
-                                    "-d", "invalid-name",
-                                    "-d", "unused-argument",
-                                    "-d", "no-self-use")
-        print("Checking prereq_tools")
-        error_count += check_script("prereq_tools",
-                                    "-d", "too-many-lines",
-                                    "-d", "unused-argument")
-        print("Checking components")
-        error_count += check_script("components")
-        print("Checking build_info")
-        error_count += check_script("build_info")
-        print("Checking check_script.py")
-        error_count += check_script("check_script")
+        check_script("SCons",
+                     "-d", "too-few-public-methods",
+                     "-d", "too-many-public-methods",
+                     "-d", "invalid-name",
+                     "-d", "unused-argument",
+                     "-d", "no-self-use")
+        check_script("prereq_tools",
+                     "-d", "too-many-lines",
+                     "-d", "unused-argument")
+        check_script("components")
+        check_script("build_info")
+        check_script("check_script")
 
     if args.fname:
-        error_count += check_script(args.fname, wrap=args.wrap,
-                                    P3=args.P3)
+        for fname in args.fname:
+            if args.exclude and fname.startswith(args.exclude):
+                continue
+            check_script(fname, wrap=args.wrap)
 
-    os.unlink(pylint_rc)
     os.unlink(pylint3_rc)
 
-    if error_count:
-        sys.exit(1)
```

Comment:
Why are you removing `check_script.py`'s exiting !0 when errors are found?

Reply:
The exit codes aren't used!

the file in code_review that calls this doesn't seem to check the exit code at all, but rather look at the pylint.out file for comments.  Our code has many hundreds of pylint warnings and always has, so exiting non-zero on any warning is of no use to anyone.  This change (and others in this PR) makes it so the script will exit non-zero if there's an *error* which is something that we could meaningfully check for in the code_review script.

Reply:
Fair enough.